### PR TITLE
Enso File Watcher

### PR DESCRIPTION
**Pull request by JanBeelte**
### Pull Request Description

This PR adds a small NodeJS based tool that watches for file changes and emits events via stdout.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -852,6 +852,7 @@
 - [Add method call info for infix operators][7090]
 - [`executionComplete` response is sent on successful execution only][7143]
 - [Send info about function values][7168]
+- [Added NodeJS based tool to watch for file changes][7208]
```
Comment (wdanilo) [inline CHANGELOG.md]: 1. I believe this is not part of Enso Compiler.
2. Sentences should have dot at the end (I see other logs don't have it either - to be fixed by our Engine team, CC @JaroslavTulach).

File: app/ide-desktop/.gitignore [hunk 2]
```diff
@@ -1,3 +1,3 @@
 node_modules
-dist
+**/dist
```
Comment (wdanilo) [inline app/ide-desktop/.gitignore]: I believe that previous gitignore was ignoring "dist" in any sub-directory. I think this change is not needed.

File: app/ide-desktop/lib/file-watcher/bundle.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+/** @file Entry point for the bundler. */
```
Comment (wdanilo) [inline app/ide-desktop/lib/file-watcher/bundle.ts]: why we need `bundle.ts`? Can't we just use cmd for that? Is custom bundler needed here?

File: app/ide-desktop/lib/file-watcher/package.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    "eslint": "^8.32.0",
+    "eslint-plugin-jsdoc": "^39.6.8",
+    "typescript": "^4.9.4"
+  },
+  "optionalDependencies": {
+    "@esbuild/darwin-x64": "^0.17.15",
+    "@esbuild/linux-x64": "^0.17.15",
+    "@esbuild/windows-x64": "^0.17.15"
+  }
+}
```
Comment (wdanilo) [inline app/ide-desktop/lib/file-watcher/package.json:29]: no nl

File: app/ide-desktop/lib/file-watcher/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import yargs from 'yargs'
+import { hideBin } from 'yargs/helpers'
+import chokidar from 'chokidar'
+
+const args = yargs(hideBin(process.argv))
+    .option('path', {
+        alias: 'p',
+        type: 'string',
+        default: '.',
+        description: 'File path to watch',
```
Comment (wdanilo) [inline app/ide-desktop/lib/file-watcher/src/index.ts]: dot missing

File: app/ide-desktop/lib/file-watcher/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    })
+    .option('verbose', {
+        alias: 'v',
+        type: 'boolean',
+        default: false,
+        description:
+            'If false, only reports add, change, and unlink events.If true, reports all events.',
```
Comment (wdanilo) [inline app/ide-desktop/lib/file-watcher/src/index.ts]: missing space after dot

File: app/ide-desktop/lib/file-watcher/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+if (args.verbose) {
+    chokidar.watch(args.path).on('all', (event, path) => {
+        console.log({ event, path })
+    })
+} else {
+    chokidar
+        .watch(args.path)
+        .on('add', path => console.log({ event: 'add', path }))
+        .on('change', path => console.log({ event: 'change', path }))
+        .on('unlink', path => console.log({ event: 'unlink', path }))
```
Comment (wdanilo) [inline app/ide-desktop/lib/file-watcher/src/index.ts]: I understand that this is not finished, or is it? There should be docs describing what is the idea behind that and that the consumer should consume things from STD OUT.

File: app/ide-desktop/.gitignore (hunk 2 above)
Comment (JanBeelte) [reply app/ide-desktop/.gitignore]: true, will revert!

File: app/ide-desktop/lib/file-watcher/src/index.ts (hunk 7 above)
Comment (JanBeelte) [reply app/ide-desktop/lib/file-watcher/src/index.ts]: I believe this is complete or which events are we missing? Added some documentation!

File: app/ide-desktop/lib/file-watcher/bundle.ts (hunk 3 above)
Comment (JanBeelte) [reply app/ide-desktop/lib/file-watcher/bundle.ts]: not needed, still from my "experimentation phase", removed it.

File: CHANGELOG.md (hunk 1 above)
Comment (JanBeelte) [reply CHANGELOG.md]: Moved to IDE even though it also feels weird there.

File: app/ide-desktop/lib/file-watcher/package.json (hunk 4 above)
Comment (JanBeelte) [reply app/ide-desktop/lib/file-watcher/package.json:29]: will revert it, but this nl is added by ./run fmt

**Comment by somebody1234** [comment]
this wasn't added as a `workspace` to `app/ide-desktop/package.json`, not sure if this is intentional

**Comment by JanBeelte** [comment]
> this wasn't added as a `workspace` to `app/ide-desktop/package.json`, not sure if this is intentional

good point! @wdanilo should I add it or do we intentionally only want to build it "on demand"?

**Comment by JanBeelte** [comment]
What are the next steps here? How do we move forward?

**Comment by wdanilo** [comment]
Closing for now, as we might implement it in Java. However, in case the Java version does not work well, we will use this impl.

**Pull request closed without merging**